File: frontend/src/service/materialService.js
import axios from './axios'

export default {
  /**
   * 获取资料列表
   */
  listMaterials(params) {
    return axios.get('/api/bbs/material/list', { params })
  },

  /**
   * 获取资料详情
   */
  getMaterialDetail(id) {
    return axios.get(`/api/bbs/material/${id}`)
  },

  /**
   * 上传资料
   */
  uploadMaterial(data) {
    return axios.post('/api/bbs/material/upload', data)
  },

  /**
   * 更新资料
   */
  updateMaterial(id, data) {
    return axios.put(`/api/bbs/material/${id}`, data)
  },

  /**
   * 更新资料状态
   */
  updateStatus(id, status) {
    return axios.put(`/api/bbs/material/${id}/status`, null, {
      params: { status }
    })
  },

  /**
   * 删除资料
   */
  deleteMaterial(id) {
    return axios.delete(`/api/bbs/material/${id}`)
  },

  /**
   * 我的资料(卖家)
   */
  myMaterials(pageNum, pageSize = 10) {
    return axios.get('/api/bbs/material/my-materials', {
      params: { pageNum, pageSize }
    })
  },

  /**
   * 已购买的资料
   */
  purchasedMaterials(pageNum, pageSize = 10) {
    return axios.get('/api/bbs/material/purchased', {
      params: { pageNum, pageSize }
    })
  },

  /**
   * 搜索资料
   */
  searchMaterials(keyword, categoryId, pageNum, pageSize = 10) {
    return axios.get('/api/bbs/material/search', {
      params: { keyword, categoryId, pageNum, pageSize }
    })
  },

  /**
   * 获取分类列表
   */
  getCategories() {
    return axios.get('/api/bbs/material/categories')
  },

  /**
   * 检查访问权限
   */
  checkAccess(materialId) {
    return axios.get(`/api/bbs/material/${materialId}/check-access`)
  },

  /**
   * 获取用户积分
   */
  getUserCredits() {
    return axios.get('/api/user/credits')
  },

  /**
   * 获取评价列表
   */
  getReviews(params) {
    return axios.get('/api/bbs/material/review/list', { params })
  },

  /**
   * 提交评价
   */
  submitReview(data) {
    return axios.post('/api/bbs/material/review', data)
  },

  /**
   * 获取相似资料
   */
  getSimilarMaterials(materialId) {
    return axios.get(`/api/bbs/material/${materialId}/similar`)
  },

  /**
   * 获取热门资料
   */
  getHotMaterials(limit = 10) {
    return axios.get('/api/bbs/material/hot', {
      params: { limit }
    })
  },

  /**
   * 获取推荐资料
   */
  getRecommendMaterials(limit = 10) {
    return axios.get('/api/bbs/material/recommend', {
      params: { limit }
    })
  },

  /**
   * 增加浏览次数
   */
  incrementViewCount(materialId) {
    return axios.post(`/api/bbs/material/${materialId}/view`)
  }
}

File: legacy/java-material/MaterialService.java
package com.liang.bbs.material.facade.service;

import com.bbs.common.response.PageResult;
import com.liang.bbs.material.facade.dto.*;

import java.util.List;

/**
 * 资料服务接口
 *
 */
public interface MaterialService {

    /**
     * 上传资料
     */
    MaterialDTO uploadMaterial(MaterialUploadRequest request);

    /**
     * 更新资料信息
     */
    void updateMaterial(Long id, MaterialDTO dto);

    /**
     * 更新资料状态
     */
    void updateStatus(Long id, Integer status);

    /**
     * 获取资料详情
     */
    MaterialDTO getById(Long id);

    /**
     * 分页查询资料列表
     */
    PageResult<MaterialDTO> listMaterials(MaterialQueryRequest request);

    /**
     * 我的资料（卖家）
     */
    PageResult<MaterialDTO> getMyMaterials(Long userId, Integer pageNum, Integer pageSize);

    /**
     * 我购买的资料
     */
    PageResult<MaterialDTO> getPurchasedMaterials(Long userId, Integer pageNum, Integer pageSize);

    /**
     * 删除资料
     */
    void deleteMaterial(Long id);

    /**
     * 搜索资料
     */
    PageResult<MaterialDTO> searchMaterials(String keyword, Long categoryId, Integer pageNum, Integer pageSize);

    /**
     * 获取分类列表
     */
    List<MaterialCategoryDTO> getCategories();

    /**
     * 获取热门资料
     */
    List<MaterialDTO> getHotMaterials(Integer limit);

    /**
     * 获取推荐资料
     */
    List<MaterialDTO> getRecommendMaterials(Long userId, Integer limit);

    /**
     * 增加浏览次数
     */
    void incrementViewCount(Long materialId, Long userId);

    /**
     * 检查用户是否有访问权限
     */
    boolean checkAccess(Long materialId, Long userId);
}
